eng/tools/spec-gen-sdk-runner/src/spec-helpers.ts

```diff
@@ -182,3 +183,180 @@ export function detectChangedSpecConfigFiles(commandInput: SpecGenSdkCmdInput):
 
   return changedSpecs;
 }
+
+/**
+ * Grouping spec configs by service based on the provided TypeSpec configs and readme files.
+ * @param tspconfigs - Array of TypeSpec config file paths.
+ * @param readmes - Array of readme file paths.
+ * @param skipUnmatchedReadme - Flag to skip unmatched readme files.
+ * @returns An array of SpecConfigs objects containing the paths to the readme and TypeSpec config files.
+ */
+export function groupSpecConfigPaths(
+  tspconfigs?: string[],
+  readmes?: string[],
+  skipUnmatchedReadme: boolean = false,
+): SpecConfigs[] {
+  const emptyArray: string[] = [];
+  const safeTspConfigs = tspconfigs ?? emptyArray;
+  const safeReadmes = readmes ?? emptyArray;
+
+  if (skipUnmatchedReadme) {
+    // Explicitly remove the specific readme file from the array
+    // as we don't want to include it for typespec related batch runs
+    const indexToRemove = safeReadmes.indexOf(
+      "specification/awsconnector/resource-manager/readme.md",
```

Comment:
Probably worth adding a comment as to why this is removed.

Reply:
The issue here is 'awsconnector' service folder contains a bunch of sub services and all of them don't enable SDK emitters in their 'tspconfig.yaml'. This causes every sub service to generate code based on parent readme.md.
With further thoughts, I plan to add an option to 'spec-gen-sdk' tool to skip SDK generation for TypeSpecs batch runs if the paired 'readme.md' has the SDK configuration but 'tspconfig.yaml' don't have the SDK configuration.